OpenRA.Mods.Common/Traits/Mobile.cs

```diff
@@ -176,6 +175,8 @@ public MovementType CurrentMovementTypes
 		INotifyFinishedMoving[] notifyFinishedMoving;
 		IWrapMove[] moveWrappers;
 		bool requireForceMove;
+		public bool TurnToMove;
+		public bool IsBlocking;
```

Comment:
Done for `IsBlocking`.

`TurnToMove` is referenced by `Mobile.IsLeaving()` so making it a part of `Move` would mean that `Mobile` would need to know about the current activity state. It seems preferable to have activities reference the internals of traits than the other way around.

Reply:
What if `IsLeaving` wasn't defined on `Mobile`? What if it was just a private helper in `Move`? It's the only thing interested in the notion of `Leaving` anyway. Then there's no poking at internals from either side?

Unless there's some semantic reason to keep the "I'm turning as part of a move" flag visible from the trait? I'm only thinking in narrow "Let's not have public get/set properties here since it seems like some private fields would suffice" and maybe there's a wider picture I'm missing.

Reply:
`IsLeaving` is called by the `Move` activity of the _other_ actor. Making it part of `Move` would mean that an activity on one actor would need to dig in the internals of a different actors activity state. That would be pretty much the worst case scenario in terms of keeping internals internal. `Mobile` is acting like an intermediary here, to provide an interface that other units can access.

Reply:
I knew I was missing something - this makes much more sense now, thanks.